### PR TITLE
Refactor: remove portal

### LINKED ISSUE #5: Restructure `apps` directory
Restructure the file structure in `apps`:

Wanted:
```
app/
  data/
    _alias
    DemoAppDataSource/
      models/
        DemoApplication.json (extends `system/SIMOS/Application`)
      instances/
        demoApplication.json (hardcode `<application_id>`)
  data-sources/
    DemoAppDataSource.json
```

Current:
```
❯ tree apps
apps
├── data
│   └── DemoAppDataSource
│       ├── instances
│       │   └── demoApplication.json
│       ├── models
│       │   └── DemoApplication.json
└── data_sources
    └── DemoAppDataSource.json

5 directories, 4 files
```

### LINKED ISSUE #6: Remove portal from demo-app
- [x] Add `app-settings.json` with `<DataSource>/<applicationId>`
```
src/
  app-settings.json
```
- [x] Replace `useSearch()` with `useDocument()` (retrieve app plugins)
- [x] Add `DemoApplication.json` blueprint with a `uiRecipe` named `demoApp`
- [x] Add UIPluginSelector to render application UI plugin

### PR DESCRIPTION
**Pull request by netr0m**
- Removed all references to `appSelector`
- Removed all references to `portal`

- Restructured `apps/` (ref. #5)
  - Changed `DemoApplication` blueprint:
    - Inherits from `system/SIMOS/Application`
    - Added `urlPath` attribute
    - Added default uiRecipe: `demoApp`
  - Renamed datasource to use consistent name: `DemoDS => DemoApplicationDataSource`

- Now loads `<dataSourceId>/<applicationId>` from `app-settings.json`
  - Fetches app settings from DMSS (entity `demoApplication.json`, from `app/data/DemoApplicationDataSource/instances/`)

- Now uses `UIPluginSelector` to load the UiPlugin for the application based on the application settings (`DemoApplicationDataSource/DemoApplication`)

- Linting:
  - Avoid 'unnamed default exports', e.g. `export default []` => `const x = []; export default x`

- Resolves #6 
- Resolves #5

### CODE REVIEW

**Review by eoaksnes — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -35,143 +29,68 @@ const GlobalStyle = createGlobalStyle`
...
+      </div>
     )
+  }
+
+  return (
+    <ThemeProvider theme={theme}>
+      <div>
+        <GlobalStyle />
+        <Switch>
+          <Route
```
Comment (soofstad) [inline src/App.tsx:70]: I don't think there should be any "Routes" in here. If an app needs different routes, thats a part of that app.
This "Bootstrapper" should just render the "app plugin" directly.

File: src/app-settings.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "dataSourceId": "DemoApplicationDataSource",
+  "applicationId": "4483c9b0-d505-46c9-a157-94c79f4d7a6a"
```
Comment (soofstad) [inline src/app-settings.json:3]: These should be combined into one "absolute ref".
The dataSourceId should not be needed for anything else than fetching this first AppSettings entity.

**Review by soofstad — CHANGES_REQUESTED** [review]

**Review by soofstad — APPROVED** [review]

**Comment by netr0m** [comment]
- Awaiting equinor/dm-cli#13

**Review by eoaksnes — APPROVED** [review]
Good work

**Review by KristianKjerstad — APPROVED** [review]

**Pull request merged**